packages/gatsby-plugin-gatsby-cloud/src/components/buttons/InfoIndicatorButton.js

```diff
@@ -1,117 +1,225 @@
+            return {
+              ...btnProps,
+              tooltip: {
+                testId: btnProps.testId,
+                content: (
+                  <FeedbackTooltipContent
+                    url={url}
+                    onOpened={() => {
+                      closeFeedbackTooltip()
+                    }}
+                  />
+                ),
+                overrideShow: true,
+                closable: true,
+                onClose: closeFeedbackTooltip,
+                onDisappear: onFeedbackTooltipDisappear,
+              },
+              active: true,
+              highlighted: true,
+            }
+          })
+        } else {
+          setButtonProps(btnProps => {
+            return {
+              ...btnProps,
+              tooltip: {
+                testId: btnProps.testId,
+                content: `Preview updated ${formatDistance(
+                  Date.now(),
+                  new Date(createdAt),
+                  { includeSeconds: true }
+                )} ago`,
+                overrideShow: false,
+                show: false,
+              },
+              active: true,
+            }
+          })
+        }
+      },
+      [BuildStatus.SUCCESS]: () => {
+        setButtonProps(btnProps => {
+          return {
+            ...btnProps,
+            tooltip: {
+              testId: btnProps.testId,
+              content: (
+                <BuildSuccessTooltipContent
+                  isOnPrettyUrl={isOnPrettyUrl}
+                  sitePrefix={sitePrefix}
+                  buildId={buildId}
+                  siteId={siteId}
+                  orgId={orgId}
+                />
+              ),
+              closable: true,
+              onClose: closeInfoTooltip,
+            },
+            active: true,
+            hoverable: true,
+          }
+        })
+      },
+      [BuildStatus.ERROR]: () => {
+        setButtonProps(btnProps => {
+          return {
+            ...btnProps,
+            tooltip: {
+              testId: btnProps.testId,
+              content: (
+                <BuildErrorTooltipContent
+                  siteId={siteId}
+                  orgId={orgId}
+                  buildId={erroredBuildId}
+                />
+              ),
+              closable: true,
+              onClose: closeInfoTooltip,
+            },
+            active: true,
+            hoverable: true,
+          }
+        })
+      },
+      [BuildStatus.BUILDING]: () => {
+        setButtonProps(btnProps => {
+          return {
+            ...btnProps,
+            tooltip: {
+              testId: btnProps.testId,
+              content: `Building a new preview`,
+              overrideShow: true,
+            },
+            active: false,
+            hoverable: true,
+            showSpinner: true,
+          }
+        })
+      },
+    }
+    const buildStatusAction = buildStatus
+      ? buildStatusActions[buildStatus]
+      : null
```

Comment:
No need to change this but fyi you can make it more concise like `buildStatusActions?.[buildStatus] || null`